### PR TITLE
test: update fcm sample app with 3rd party SDK that handles push

### PR DESCRIPTION
**Pull request by levibostian**
Setup fcm sample app to be able to test new iOS feature: https://github.com/customerio/issues/issues/11289

# Steps to make a local build

* On root directory of project, run `npm install` 
* navigate to `Apps/FCM/` and run `npm install`. 
* Open `Apps/FCM/ioss/` directory in Xcode. You may need to download code signing files, create `Env.swift` file with CIO credentials.

### CODE REVIEW

File: Apps/FCM/ios/FCMSampleApp/AppDelegate.h [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #import <RCTAppDelegate.h>
 #import <UIKit/UIKit.h>
-#import <UserNotifications/UserNotifications.h>
 #import <FirebaseMessaging/FIRMessaging.h>
 
-@interface AppDelegate : RCTAppDelegate <UNUserNotificationCenterDelegate, FIRMessagingDelegate>
+@interface AppDelegate : RCTAppDelegate <FIRMessagingDelegate>
```
Comment (levibostian) [inline Apps/FCM/ios/FCMSampleApp/AppDelegate.h:5]: Removed parts of code the RN CIO SDK no longer requires.

File: Apps/FCM/ios/FCMSampleApp/AppDelegate.mm [hunk 2]
```diff
@@ -33,7 +33,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     }
   }
   
-  [pnHandlerObj setupCustomerIOClickHandling:self];
+  [pnHandlerObj setupCustomerIOClickHandling];
```
Comment (levibostian) [inline Apps/FCM/ios/FCMSampleApp/AppDelegate.mm:36]: Removed parts of code the RN CIO SDK no longer requires.

File: Apps/FCM/ios/FCMSampleApp/Info.plist [hunk 3]
```diff
@@ -50,6 +50,11 @@
 	</dict>
 	<key>NSLocationWhenInUseUsageDescription</key>
 	<string></string>
+	<key>UIBackgroundModes</key>
```
Comment (levibostian) [inline Apps/FCM/ios/FCMSampleApp/Info.plist:53]: rn-firebase's docs request that these background modes are enabled in app.

File: Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift [hunk 4]
```diff
@@ -9,8 +8,8 @@ public class MyAppPushNotificationsHandler : NSObject {
 
   public override init() {}
 
-  @objc(setupCustomerIOClickHandling:)
-  public func setupCustomerIOClickHandling(withNotificationDelegate notificationDelegate: UNUserNotificationCenterDelegate) {
+  @objc(setupCustomerIOClickHandling)
```
Comment (levibostian) [inline Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift:11]: Changed this file to what RN customers will begin using when the iOS feature launches.

File: Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift [hunk 5]
```diff
@@ -21,26 +20,12 @@ public class MyAppPushNotificationsHandler : NSObject {
       // They are optional for your setup. 
       config.logLevel = .debug
     }
-    
-    let center  = UNUserNotificationCenter.current()
-    center.delegate = notificationDelegate
+    MessagingPushFCM.initialize(configOptions: nil)
```
Comment (Shahroz16) [inline Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift:23]: same as last PR, can we please add that this is mandatory?

**Review by Shahroz16 — APPROVED** [review]

**Review by ami-aman — APPROVED** [review]

File: Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift (hunk 5 above)
Comment (levibostian) [reply Apps/FCM/ios/FCMSampleApp/MyAppPushNotificationsHandler.swift:23]: After [this conversation is resolved](https://github.com/customerio/customerio-reactnative/pull/225#discussion_r1441769902), I'll act upon this one.

File: Apps/FCM/package.json [hunk 6]
```diff
@@ -11,6 +11,8 @@
   "dependencies": {
     "@react-native-async-storage/async-storage": "^1.19.1",
     "@react-native-clipboard/clipboard": "^1.11.2",
+    "@react-native-firebase/app": "^18.7.2",
```
[the remark this answers is not in the document]
Comment (levibostian) [reply Apps/FCM/package.json:14]: [The rn-firebase docs do ask you to, yes](https://rnfirebase.io/#1-install-via-npm)

**Pull request merged**